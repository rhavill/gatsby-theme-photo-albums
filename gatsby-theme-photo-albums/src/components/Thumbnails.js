import React from 'react'
import PropTypes from 'prop-types'
import Thumbnail from './Thumbnail'
import {mapIndexed} from '../util/ramda-utils'
import {getPhotoPathWithPage} from '../util/photo-paths'

const Thumbnails = ({files, currentPage}) => {
  return mapIndexed((file, i) => 
    <Thumbnail key={i} path={getPhotoPathWithPage(currentPage, file.url)} 
      imageData={file.imageData} />
  )(files)
}

Thumbnails.propTypes = {
  currentPage: PropTypes.number.isRequired,
  files: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string.isRequired,
      imageData: PropTypes.object.isRequired,
    }).isRequired   
  ).isRequired
}

export default Thumbnails