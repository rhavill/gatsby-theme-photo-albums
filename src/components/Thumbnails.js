import React from 'react'
import PropTypes from 'prop-types'
import {graphql} from 'gatsby'
import addIndex from 'ramda/src/addIndex'
import compose from 'ramda/src/compose'
import map from 'ramda/src/map'
import prop from 'ramda/src/prop'
import Thumbnail from './Thumbnail'
import getChildPaths from '../util/source-filesystem-child-paths'

const mapIndexed = addIndex(map)

const Thumbnails = ({path, data}) => 
  compose(
    mapIndexed((folder, i) => 
      <Thumbnail key={i} path={folder} imageData={data.photos.nodes[i]} />
    ),
    getChildPaths(path),
    map(prop('relativePath'))
  )(data.photos.nodes)

export const query = graphql`
  fragment ThumbnailsFragment on Query {
    photos: allFile(filter: {relativePath: {ne: "folder.png"}} limit: $limit skip: $skip) {
      nodes {
        relativePath
        ...ThumbnailFragment
      }
    }
  }
`

Thumbnails.propTypes = {
  path: PropTypes.string.isRequired,
  data: PropTypes.shape({
    photos: PropTypes.shape({
      nodes: PropTypes.arrayOf(
        PropTypes.shape({
          relativePath: PropTypes.string.isRequired
        }).isRequired   
      ).isRequired
    }).isRequired  
  }).isRequired
}

export default Thumbnails