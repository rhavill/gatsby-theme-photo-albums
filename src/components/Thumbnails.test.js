import React from 'react'
import renderer from 'react-test-renderer'
import {cleanup} from '@testing-library/react';
import Thumbnails from './Thumbnails'
import fileData from '../test-data/source-filesystem-file-data'

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

describe("Thumbnails", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Thumbnails path='/level-one' data={fileData} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})