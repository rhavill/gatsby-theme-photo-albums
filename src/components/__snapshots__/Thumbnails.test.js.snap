// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnails renders correctly 1`] = `
<article
  className="file"
  data-testid="file"
>
  <a
    href="/level-one/2.jpg"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "display": "inline-block",
          "height": 250,
          "overflow": "hidden",
          "position": "relative",
          "width": 250,
        }
      }
    >
      <picture>
        <source
          srcSet="/static/blah/blah/1.jpg 1x"
        />
        <img
          alt="2"
          height={250}
          loading="lazy"
          onLoad={[Function]}
          src="/static/blah/blah/1.jpg"
          srcSet="/static/blah/blah/1.jpg 1x"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
          title="2"
          width={250}
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"/static/blah/blah/1.jpg 1x\\" /><img loading=\\"lazy\\" width=\\"250\\" height=\\"250\\" srcset=\\"/static/blah/blah/1.jpg 1x\\" src=\\"/static/blah/blah/1.jpg\\" alt=\\"2\\" title=\\"2\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </a>
</article>
`;
